packages/ui/components/form/step/steps.test.tsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable playwright/missing-playwright-await */
+import { render, fireEvent } from "@testing-library/react";
+import { vi } from "vitest";
+
+import { Steps } from "./Steps";
+
+const MAX_STEPS = 4;
+let CURRENT_STEP = 2;
+const mockNavigateToStep = vi.fn();
+
+const Props = {
+  maxSteps: MAX_STEPS,
+  currentStep: CURRENT_STEP,
+  navigateToStep: mockNavigateToStep,
+  stepLabel: (currentStep: number, totalSteps: number) => `Test Step ${currentStep} of ${totalSteps}`,
+};
+
+describe("Tests for Steps Component", () => {
+  test("Should render the correct number of steps", () => {
+    const { getByTestId, getByText } = render(<Steps {...Props} />);
+
+    for (let i = 0; i < Props.maxSteps; i++) {
+      const step = getByTestId(`step-indicator-${i}`);
```

Comment:
`getByTestId` gets by data-testid attribute only. So, it would always have` toHaveAttribute("data-testid")` assertion passing.
I don't think we need `toHaveAttribute` assertion, instead we should check that the steps are clickable. There are 2 things in there
- Presence of` cursor-pointer` class. I know that we shouldn't test implementation details but this class is something that must always be present as long as we are using tailwind
- Clicking every step till current step and checking that navigateToStep is being called.